feat(linter/eslint): show ignore patterns in `eslint/no-unused-vars` diagnostic messages

DonIsaac opened the pull request:
Inform users what pattern an unused variable must match in diagnostic help messages.

```
  ⚠ eslint(no-unused-vars): Variable 'a' is declared but never used. Unused variables should start with a '_'.
   ╭─[no_unused_vars.tsx:1:5]
 1 │ var a=10
   ·     ┬
   ·     ╰── 'a' is declared here
   ╰────
  help: Consider removing this declaration.
```

DonIsaac commented:
* **#6697** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6697?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6696** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6696?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

camchenry commented on crates/oxc_linter/src/rules/eslint/no_unused_vars/options.rs:
```diff
@@ -271,6 +271,22 @@ impl<R> IgnorePattern<R> {
+        match self {
+            Self::None => Cow::Borrowed(""),
+            Self::Default => {
+                Cow::Owned(format!(" Unused {symbol_kind_plural} should start with a '_'."))
```
Minor: we could drop the extra letter to make it a bit shorter. 
```suggestion
                Cow::Owned(format!(" Unused {symbol_kind_plural} should start with '_'."))
```

camchenry commented on crates/oxc_linter/src/rules/eslint/no_unused_vars/diagnostic.rs:
```diff
@@ -1,61 +1,95 @@
+    let suffix = pat.diagnostic_help(pronoun_plural);
 
-    OxcDiagnostic::warn(format!("{pronoun} '{name}' is {verb} but never used."))
+    OxcDiagnostic::warn(format!("{pronoun} '{name}' is {verb} but never used.{suffix}"))
```
Would it make more sense to put this in the help text? It seems like it would make more sense there. I wish we could supply multiple help messages.

DonIsaac replied:
I tend to agree. It would be nice to support multiple help messages. This follows ESLint's message, so I'm going to leave it here for now.

DonIsaac committed:
feat(linter/eslint): show ignore patterns in `eslint/no-unused-vars` diagnostic messages (#6696)

Inform users what pattern an unused variable must match in diagnostic help messages.

```
  ⚠ eslint(no-unused-vars): Variable 'a' is declared but never used. Unused variables should start with a '_'.
   ╭─[no_unused_vars.tsx:1:5]
 1 │ var a=10
   ·     ┬
   ·     ╰── 'a' is declared here
   ╰────
  help: Consider removing this declaration.
```
